wagtail/documents/views/chooser.py

```diff
@@ -151,8 +165,21 @@ def chooser_upload(request):
 
     documents = Document.objects.order_by('title')
 
+    for hook in hooks.get_hooks('construct_document_chooser_queryset'):
```

Comment:
Wondering if we could merge this in to ``get_chooser_context()`` as well as the search/pagination code from the ``chooser`` view so this isn't repeated?

Reply:
I think so. Actually, the same changes can be made to the images chooser. Is it better to make changes in this pull request or create another one after the documents chooser will be updated?

Reply:
It's slightly nicer to have them in the same PR so that changes to both get merged at the same time